### PR TITLE
SALTO-2863 Add E2E test to check empty field deploy

### PR DESCRIPTION
**Pull request by Uri-Bar**
_Add E2E test to check empty field deploy_

---
_Additional context for reviewer_:
None

---
_Release Notes_: 
Salesforce-adapter:

* Add E2E test to check empty field deploy
---

_User Notifications_: 
Salesforce-adapter:

None

### CODE REVIEW

File: packages/adapter-utils/src/utils.ts [hunk 1]
```diff
@@ -514,7 +514,7 @@ const isEqualResolvedValues = (
 )
 
 export const restoreValues: RestoreValuesFunc = async (
-  source, targetElement, getLookUpName, allowEmpty = false
```
Comment (Uri-Bar) [inline packages/adapter-utils/src/utils.ts:517]: @dantal4 @alonstern @netama  we find another place in adapter utils that has allowEmpty = true that affecting SF, would like your approval that this change is not affecting NS, Jira or Zendesk

Comment (dantal4) [reply packages/adapter-utils/src/utils.ts:517]: no effect on netsuite.

Comment (netama) [reply packages/adapter-utils/src/utils.ts:517]: please make sure this is covered in unittests inside `adapter-utils`, there seem to already be tests both for `restoreValues` and for `restoreChangeElement` so if this the correct behavior from now on we should make sure it is only changed intentionally

Comment (alonstern) [reply packages/adapter-utils/src/utils.ts:517]: Looks good from Jira side (if anything it fixes the places where we use the restore functions)

**Review by tamtamirr — APPROVED** [review]
LGTM :) thanks for this
